### PR TITLE
#184629384 user signup with validations 

### PR DESCRIPTION
**Pull request by blackd44**
#### What does this PR do?
- creating account for a new user and validate his/her information
- testing the user signup routes
- adding documentation of the user signup route

#### Description of Task to be completed?
user must enter the data such as
- username
- email
- password
- gender

then its validate those information
then check if there isn't other account with that email address
and add it in the database 
after return the message that user has been created

endpoints:
- /users/signup

#### How should this be manually tested?
after cloning the repo,
- in terminal add `npm install`
- add `.env` in your repo and assign the variables as there are in `.env.example` file
- run in termainal `npm run migrate:all` to setup the database
- run in termainal `npm start` to start the project
- in your browser add `http://localhost:<your port>/api-docs` for documentantion

#### Any background context you want to provide?
n/a

#### What are the relevant pivotal tracker/Trello stories?
https://www.pivotaltracker.com/n/projects/2632830/stories/184629384

#### Screenshots (if appropriate)
Screen Shot 2023-03-09 at 13 15 32
Screen Shot 2023-03-09 at 14 31 00

#### Questions:
n/a

### CODE REVIEW

**Review by Zeden12 — APPROVED** [review]
LGTM

File: src/controllers/users.controllers.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+
+export default class Users {
+  static async signup(req, res) {
+    try {
+      let { data } = await User.register(req.body)
+      return res.status(200).json({ message: "user created successful", data});
+    }
+    catch(e) {
+      console.log(e)
```
Comment (nshutijonathan) [inline src/controllers/users.controllers.js]: remove console.log, it should be for debugging purposes only, otherwise it should not be pushed

**Review by nshutijonathan — CHANGES_REQUESTED** [review]
Please attach a screenshot in the PR showing your test coverage.

**Review by nshutijonathan — CHANGES_REQUESTED** [review]
Also, update the readme, I would like to know how as a non-technical person I can pass those values in postman while signing up. Only the format of those JSON values and the endpoint

Also, your eslint is claiming, so please take a look at that.

**Review by bencyubahiro77 — APPROVED** [review]
check the Eslint feedback and you are good to go

**Comment by blackd44** [comment]
> Also, update the readme, I would like to know how as a non-technical person I can pass those values in postman while signing up. Only the format of those JSON values and the endpoint
> 
> Also, your eslint is claiming, so please take a look at that.

i think it was fixed

**Review by edwinnambaje — APPROVED** [review]
LGTM

**Review by Habihirwe — APPROVED** [review]
great job !

**Review by nshutijonathan — APPROVED** [review]
LGTM

**Pull request merged**